### PR TITLE
redpanda: unbreak client and mark server as broken

### PR DESCRIPTION
**Pull request by happysalada**
## Description of changes

in a previous PR the redpanda client was erroneously marked as broken. This unbreaks the client and correctly marks the server as broken.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -12615,9 +12615,9 @@ with pkgs;
 
   redmine = callPackage ../applications/version-management/redmine { };
 
-  redpanda = callPackage ../servers/redpanda { };
+  redpanda-client = callPackage ../servers/redpanda { };
```
Comment (OPNA2608) [inline pkgs/top-level/all-packages.nix:12618]: Should prolly get an entry to `pkgs/top-level/aliases.nix`.

**Comment by OPNA2608** [comment]
And commit title uses no-longer-existing `redpanda` instead of the renamed `redpanda-client` so OfBorg doesn't know what to build. Could you split the changes into separate commits and refer to the client via its new attribute name?

@ofborg build redpanda-client

**Comment by OPNA2608** [comment]
Ah, unfree license so OfBorg won't build it anyway... Still, commit title should be changed & contents split into two per-package commits IMO.

**Comment by happysalada** [comment]
@OPNA2608 just split in two commits and addressed the feedback

**Review by OPNA2608 — COMMENTED** [review]
Result of `nixpkgs-review pr 260872` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>redpanda-client</li>
  </ul>
</details>

Binary can produce help & version output.

---

The client commit title doesn't mention the un-breaking, otherwise LGTM.

**Comment by happysalada** [comment]
Thank you for the review. I dont have a lot of time at the moment so i just went ahead and merged with the non ideal commit message, i hope its okay with you.

**Pull request merged**